mypy/typeanal.py

```diff
@@ -956,20 +983,34 @@ def is_defined_type_var(self, tvar: str, context: Context) -> bool:
             return False
         return self.tvar_scope.get_binding(tvar_node) is not None
 
-    def anal_array(self, a: Iterable[Type], nested: bool = True) -> List[Type]:
+    def anal_array(self,
+                   a: Iterable[Type],
+                   nested: bool = True, *,
+                   allow_param_spec: bool = False) -> List[Type]:
         res: List[Type] = []
         for t in a:
-            res.append(self.anal_type(t, nested))
+            res.append(self.anal_type(t, nested, allow_param_spec=allow_param_spec))
         return res
 
-    def anal_type(self, t: Type, nested: bool = True) -> Type:
+    def anal_type(self, t: Type, nested: bool = True, *, allow_param_spec: bool = False) -> Type:
         if nested:
             self.nesting_level += 1
         try:
-            return t.accept(self)
+            analyzed = t.accept(self)
         finally:
             if nested:
                 self.nesting_level -= 1
+        # Use type(...) to ignore proper/non-proper type distinction.
+        if (not allow_param_spec
+                and type(analyzed) is ParamSpecType
```

Comment:
Is this really needed? You can update the plugin instead as I suggested above.

Reply:
Replaced with regular `isinstance`.